content/blog/evolution-of-inline-editing.md

```diff
@@ -0,0 +1,69 @@
+   ></iframe>
+
+We gain a lot of control as developers with this approach. The inline editing components live with your layout components and behave pretty much as you would expect them to.
+
+But there are a few trade-offs to this approach that we've [discussed before](/blog/more-changes-coming-to-inline-editing/).
+
+- Components like `InlineText` add markup and behaviour that can interfere with your desired layout and styling. You now have an `input` element and wrapping divs where you'd normally just render text. 
+- Sidebar editing forms are constructed from a single configuration object. Inline editing brings a separate developer experience and set of concepts to understand. 
+
+Another side effect of having inline editing coupled to your components is that you're now bringing TinaCMS with your components, everywhere.
+
+We could try conditionally adding the inline editing components, but it's clear that the logic needed to include inline editing is now coupled to our page and components. It's now more difficult to maintain, share, or re-use those components.
+
+The usability benefits of inline editing could outweigh these concerns depending on your use case. But we're searching for solutions with fewer trade-offs.
+
+## Alternative approaches
+
+### Ref-Based Inline Editing
+
+`useFieldRef` is the first part of an experimental new API for creating a [ref-based inline editing](https://github.com/tinacms/tinacms/blob/master/packages/react-tinacms-inline/README.md#usefieldref-ref-based-inline-editing) experience with Tina. With `useFieldRef`, inline editing components are defined in the form configuration, and you assign a [ref](https://reactjs.org/docs/refs-and-the-dom.html) to the component in your layout that the field should attach to.
+
+Inline fields created in this fashion will be absolutely positioned on top of the referenced component and conform to its dimensions. This makes it possible for the field to appear as if it were replacing the layout component in the DOM without altering the markup.
+
+In contrast to our first example, we define an `inlineComponent` key in our form configuration object. We've now added an inline editing component in the same manner as setting up sidebar editing!
+
+We then call `useFieldRef` to acquire the ref that will be attached to our title element.
+
+<iframe src="https://codesandbox.io/embed/tina-ref-based-inline-editing-p8kx4?fontsize=14&hidenavigation=1&theme=dark&view=split&editorsize=65"
+     style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;"
+     title="tina-inline-editing"
+     allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
+     sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
+   ></iframe>
+
+### Editing Routes
```

Comment:
I believe @jeffsee55 is using another pattern on the cloud starter, than only loads Tina after auth, is that worth mentioning?

Reply:
I think that's made possible by having only sidebar editing. I think it's a lot trickier with inline editing.